### PR TITLE
Add documentation for treepicker

### PR DESCRIPTION
**Pull request by nutboltu**
tree-picker-docu

### CODE REVIEW

File: docs/examples/TreePickerExample.jsx [hunk 1]
```diff
@@ -71,7 +71,198 @@ const exampleProps = {
   searchOnClear={this.pickerSearchOnClear}
   additionalClassNames={this.state.pickerSearchValue ? undefined : ['background-highlighted', 'test-class']}
 />`,
-  propTypes: [],
+  propTypes: [ {
+    propType: 'additionalClassNames',
+    type: 'arrayOf string',
+    defaultValue: <pre>-</pre>,
```
Comment (lightbringer1991) [inline docs/examples/TreePickerExample.jsx]: I reckon we can ignore this property if it doesn't actually have a value

**Comment by nutboltu** [comment]
@omgaz I am going to discard those dist files in my next changes

**Comment by omgaz** [comment]
Thanks, it looks like it's only node8 that's failing. Could be something to do with the package-lock.json in master let me take a look.

File: docs/examples/TreePickerExample.jsx [hunk 2]
```diff
@@ -71,7 +71,192 @@ const exampleProps = {
...
+  {
+    propType: 'emptySelectedListSvgSymbol',
+    type: <span>shapeOf <a href="#svg-symbol-component">SVG Symbol</a> prop types.</span>,
+    defaultValue: <pre>-</pre>,
+    note: `displays this svg symbol when there will be no item on right Grid(Selected list)`
+  },
+  {
+    propType: 'emptyText',
+    type: 'string',
+    defaultValue: <pre>-</pre>,
```
Comment (omgaz) [inline docs/examples/TreePickerExample.jsx]: If you don't define a default value the css will do this - for you so you can remove all `    defaultValue: <pre>-</pre>,` from the file

File: docs/examples/TreePickerExample.jsx [hunk 3]
```diff
@@ -71,7 +71,192 @@ const exampleProps = {
...
+    propType: 'selectedNodes',
+    type: 'arrayOf Treepicker Nodes',
+    defaultValue: <pre>-</pre>,
+    note: 'required',
+  },
+  {
+    propType: 'subtree',
+    type: 'arrayOf Treepicker Nodes',
+    defaultValue: <pre>-</pre>,
+    note: 'required. Sub list of selected nodes',
```
Comment (omgaz) [inline docs/examples/TreePickerExample.jsx]: The subtree isn't the list of selected nodes it;s a list of available unselected nodes

File: docs/examples/TreePickerExample.jsx [hunk 4]
```diff
@@ -71,7 +71,192 @@ const exampleProps = {
...
+    propType: 'searchOnChange',
+    type: 'bool',
+    defaultValue: <pre>true</pre>,
+    note: 'serach for item on onChange trigger when the value is true'
+  },
+  {
+    propType: 'searchOnEnterKey',
+    type: 'bool',
+    defaultValue: <pre>false</pre>,
+    note: 'serach for item on Enter key trigger when the value is true'
```
Comment (omgaz) [inline docs/examples/TreePickerExample.jsx]: typo in this and above. `serach` → `Search`

File: docs/examples/TreePickerExample.jsx [hunk 5]
```diff
@@ -71,7 +71,192 @@ const exampleProps = {
...
+    propType: 'expandNode',
+    type: 'func',
+    defaultValue: <pre>-</pre>,
+    note: `triggers when clicking any item in the left Grid`
+  },
+  {
+    propType: 'groupFormatter',
+    type: 'func',
+    defaultValue: <pre>-</pre>,
+    note: `this funciton use to transform keys of the list item in the left Grid`
```
Comment (omgaz) [inline docs/examples/TreePickerExample.jsx]: Typo: `funciton` → `function`

**Comment by nutboltu** [comment]
@omgaz all the requests are changed. Please review again

**Pull request merged**